pkg/rtc/transport.go

```diff
@@ -630,6 +636,43 @@ func (t *PCTransport) setICEConnectedAt(at time.Time) {
 			t.tcpICETimer = nil
 		}
 	}
+
+	if t.mayFailedICEStatsTimer != nil {
+		t.mayFailedICEStatsTimer.Stop()
+		t.mayFailedICEStatsTimer = nil
+	}
+	t.mayFailedICEStats = nil
+	t.lock.Unlock()
```

Comment:
confused by this, this seems to clear everything on ICE connected. Is that right?

If so, is the goal of this to log only not able to connect at all? I am guessing not as ICE restart is logging.

Will the logs be available if it is cleared here when ICE disconnects/fails later?

Reply:
It will not collect candidates if the ice state had connected since we have selected candidate logs, the goal of this pr is log candidate details only if the ice has never connected since we don't have logs for it now.

Reply:
We can observe it and add logs for the connected and failed case later if it is required, but not for now to avoid too much logs since the full candidates stats  could be large

Reply:
Got it.

Wondering if it is also useful to collect when failing after connecting to check if some candidate pair switch happened (i. e. some other pair is able to communicate). We can add it later if need be.